forge-gui/res/cardsfolder/upcoming/krenkos_buzzcrusher.txt

```diff
@@ -0,0 +1,13 @@
+Name:Krenko's Buzzcrusher
+ManaCost:2 R R
+Types:Artifact Creature Insect Thopter
+PT:4/4
+K:Flying
+K:Trample
+T:Mode$ ChangesZone | Origin$ Any | Destination$ Battlefield | ValidCard$ Card.Self | Execute$ DBForEach | TriggerDescription$ When CARDNAME enters the battlefield, for each player, destroy up to one nonbasic land that player controls. For each land destroyed this way, its controller may search their library for a basic land card, put it onto the battlefield tapped, then shuffle.
+SVar:DBForEach:DB$ RepeatEach | RepeatPlayers$ Player | RepeatSubAbility$ DBChoose | SubAbility$ DBDestroyAll
+SVar:DBChoose:DB$ ChooseCard | Defined$ You | ChoiceZone$ Battlefield | Amount$ 1 | Choices$ Land.nonBasic+RememberedPlayerCtrl | Optional$ True | ImprintChosen$ True | ChoiceTitle$ Choose up to one nonbasic land this player controls.
+SVar:DBDestroyAll:DB$ DestroyAll | ValidCards$ Land.nonBasic+IsImprinted | RememberLKI$ True | SubAbility$ DBSearch
+SVar:DBSearch:DB$ ChangeZone | Origin$ Library | Destination$ Battlefield | Tapped$ True | ChangeType$ Land.Basic | ChangeNum$ 1 | RememberChanged$ True | DefinedPlayer$ ImprintedController | Optional$ True | SubAbility$ DBCleanup
```

Comment:
why are you using ImprintedController here?
only RememberedController would have fewer (in case some were indestructible)
(the RememberChanged also does nothing)

Reply:
I'm using `ImprintedController` because I lifted a good deal of the script from [From the Ashes](https://scryfall.com/card/c13/108/from-the-ashes). I imagine then that one also needs some updating. My tendency, navigating the code, likely unaware whether or not parameters are interchangeable, is to use what, given the name, seems to be the most relevant one.
(Forgot to remove `RememberChanged` even knowing it was setting up library shuffling. My bad on that one.)

Reply:
Just replaced `ImprintedController` with `RememberedController` to test it out and I get no prompt to search my library for a land nor do tapped lands show up on my AI opponents' battlefields. I do vaguely recall this being an issue but I'm about 10 iterations deep on this script at this point.

Reply:
well RememberLKI doesn't exist on DestroyAll
so use RememberDestroyed